pkg/daemon/manager/manager.go

```diff
@@ -98,6 +115,26 @@ func (m *manager) EnsureServiceState(service winsvc.Service, state svc.State) er
 			return err
 		}
 	case svc.Stopped:
+		// Before we can stop this service, we need to make sure all services dependent on this service are stopped
+		// The service must be cast to the actual type so we can get its handle
+		winSvc, ok := service.(*mgr.Service)
+		if !ok {
+			return fmt.Errorf("service is not correct type")
+		}
+		dependentServices, err := m.listDependentServices(winSvc.Handle)
+		if err != nil {
+			return errors.Wrap(err, "error finding dependent services")
+		}
+		for _, dependentServiceName := range dependentServices {
+			dependentSvc, err := m.OpenService(dependentServiceName)
+			if err != nil {
+				return fmt.Errorf("error opening dependent service %s", dependentServiceName)
+			}
+			err = m.EnsureServiceState(dependentSvc, svc.Stopped)
```

Comment:
Is it possible to have interdependent services? While it will not happen with services we configuring but I am wondering about the BYOH case where the customer could set some service of theirs dependent on one of the WICD managed services and vice versa.